From 1fe9591d1fd6abb7c3f4a51f4518c8239a7ab75a Mon Sep 17 00:00:00 2001
From: Julian Brown <julian.brown@cpanel.net>
Date: Tue, 16 Nov 2021 16:53:31 +0000
Subject: [PATCH] Bug: Used wrong ptr value in memcpy

Submitted PR upstream, the moment they merge it and publish we need
to drop this patch.

The pull request is:

https://github.com/php/php-src/pull/7660
---
 main/fopen_wrappers.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/main/fopen_wrappers.c b/main/fopen_wrappers.c
index 59c20c82..f6ce26e1 100644
--- a/main/fopen_wrappers.c
+++ b/main/fopen_wrappers.c
@@ -402,7 +402,7 @@ PHPAPI int php_fopen_primary_script(zend_file_handle *file_handle)
 		IS_ABSOLUTE_PATH(PG(doc_root), length)) {
 		size_t path_len = strlen(path_info);
 		filename = zend_string_alloc(length + path_len + 2, 0);
-		memcpy(filename, PG(doc_root), length);
+		memcpy(ZSTR_VAL(filename), PG(doc_root), length);
 		if (!IS_SLASH(ZSTR_VAL(filename)[length - 1])) {	/* length is never 0 */
 			ZSTR_VAL(filename)[length++] = PHP_DIR_SEPARATOR;
 		}
-- 
2.31.1

